.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,358 @@
+        image: datadog/agent:latest
+        env:
+          DD_APM_ENABLED: true
+          DD_BIND_HOST: "0.0.0.0"
+          DD_API_KEY: "invalid_key_but_this_is_fine"
+        # We need to specify a custom health-check. By default, this container will remain "unhealthy" since 
+        # we don't fully configure it with a valid API key (and possibly other reasons)
+        # This command just checks for our ability to connect to port 8126
+        options: >-
+          --health-cmd "bash -c '</dev/tcp/127.0.0.1/8126'"
+        ports:
+          - 8125:8125/udp
+          - 8126:8126
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          repository: 'DataDog/dd-trace-go'
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+          check-latest: true
+          cache: true
+      - name: Install gotestsum
+        run: go install gotest.tools/gotestsum@latest
+      - name: Test Core
+        run: |
+            mkdir -p $TEST_RESULTS
+            PACKAGE_NAMES=$(go list ./... | grep -v /contrib/)
+            gotestsum --junitfile ${TEST_RESULTS}/gotestsum-report.xml -- $PACKAGE_NAMES -v -race -coverprofile=coverage.txt -covermode=atomic
+
+      - name: Upload the results to Datadog CI App
+        if: always()
+        uses: ./.github/actions/dd-ci-upload
+        with:
+          dd-api-key: ${{ secrets.DD_CI_API_KEY }}
+          files: ${{ env.TEST_RESULTS }}/gotestsum-report.xml
+          tags: go:${{ matrix.go-version }},arch:${{ runner.arch }},os:${{ runner.os }},distribution:${{ runner.distribution }}
+      - name: Upload Coverage
+        if: always()
+        shell: bash
+        run: bash <(curl -s https://codecov.io/bash)
+
+  test-multi-os:
+    needs:
+      - runner-indexes
+    strategy:
+      matrix:
+        #runs-on: [ macos-latest, windows-latest, ubuntu-latest ]
+        #go-version: ["1.16", "1.17", "1.18"]
+        runs-on: [ windows-latest ]
+        go-version: ["1.18"]
+        runner-index: ${{ fromjson(needs.runner-indexes.outputs.json) }}
+      fail-fast: false
+    runs-on: ${{ matrix.runs-on }}
+    env:
+       REPORT: gotestsum-report.xml # path to where test results will be saved
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          repository: 'DataDog/dd-trace-go'
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+          check-latest: true
+          cache: true
+      - name: Install gotestsum
+        run: go install gotest.tools/gotestsum@latest
+      - name: Mac OS Coreutils
+        if: matrix.runs-on == 'macos-latest'
+        run: brew install coreutils
+      - name: "Runner ${{ matrix.runner-index }}: Test Core and Contrib (No Integration Tests)"
+        shell: bash
+        run: |
+            go list ./... | grep -v -e grpc.v12 -e google.golang.org/api -e sarama -e confluent-kafka-go | sort >packages.txt
```

Comment:
I notice that the `profiler/internal/cmemprof` package tests fail to build on Windows since I didn't add any checks in the code to disable building for Windows. I can add those checks, but for now I think it's okay to just skip `cmemprof` entirely on Windows.

Reply:
To clarify, I was even thinking something like this: 

```diff
- go list ./... | grep -v -e grpc.v12 -e google.golang.org/api -e sarama -e confluent-kafka-go | sort >packages.txt
+ go list ./... | grep -v -e grpc.v12 -e google.golang.org/api -e sarama -e confluent-kafka-go -e cmemprof | sort >packages.txt
```

Reply:
👍 I was just adding go:build comments to it to test. I can remove those and let you do it since I don't quite know how you want it done for the profiler.

I'll exclude it for now.